akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberEntitiesShardStore.scala

```diff
@@ -93,14 +96,17 @@ private[akka] final class EventSourcedRememberEntitiesShardStore(
         (if (started.nonEmpty) EntitiesStarted(started) :: Nil else Nil) :::
         (if (stopped.nonEmpty) EntitiesStopped(stopped) :: Nil else Nil)
       var left = events.size
-      persistAll(events) { _ =>
-        left -= 1
-        if (left == 0) {
-          sender() ! RememberEntitiesShardStore.UpdateDone(started, stopped)
-          state.copy(state.entities.union(started).diff(stopped))
-          saveSnapshotWhenNeeded()
-        }
-      }
+      events
+        .grouped(maxUpdatesPerWrite)
+        .foreach(group =>
+          persistAll(group) { _ =>
+            left -= 1
+            if (left == 0) {
+              sender() ! RememberEntitiesShardStore.UpdateDone(started, stopped)
+              state.copy(state.entities.union(started).diff(stopped))
+              saveSnapshotWhenNeeded()
+            }
+          })
```

Comment:
Could perhaps avoid some overhead when the write does not need splitting up.

Reply:
👍 which often will be the case